livekit_models.proto

```diff
@@ -169,3 +169,48 @@ message ClientInfo {
   string browser_version = 8;
   string address = 9;
 }
+
+// server provided client configuration
+message RTCClientConfiguration {
+	AudioConfiguration audio = 1;
+  VideoConfiguration video = 2;
+  SignalConfiguration signal = 4;
+}
+
+message AudioConfiguration {
+  message AudioProcessing {
+    // echo cancellation
+    bool aec = 1;
+    // noise suppression
+    bool ns = 2;
+    // automatic gain control
+    bool agc = 3;
+  }
+  message AudioPreferences {
+    bool dtx = 1;
+    int32 sample_rate = 2;
+    int32 channel_count = 3;
+  }
+  Codec codec = 1;
+  AudioProcessing processing = 2;
+  AudioPreferences preferences = 3;
+
+  // TODO sample/playout config(micphone, speaker, earphone ...)
+}
+
+message VideoConfiguration {
+  message VideoProcessing {
+    bool hardware_decoder = 1;
+    bool hardware_encoder = 2;
```

Comment:
I am guessing this may need to be flipped. Have `software_*` here. Most probably there are going to be issues with hardware encoder in some combinations and we would want to use software encoding/decoding in such cases. As protobuf will not send anything on the wire for default value, we will not be able signal to turn off hardware encoder.

Reply:
good point!
afaik the `hardware_acceleration` flag is a browser wide setting and nothing we can turn on/off from JS within a browser.

Reply:
@lukasIO @boks1971 Good point! For this reason, it would be better to use option always change common behavior,  so use disable_xxx for other options like aec/ns too instead of enable it explicit.

Reply:
from reading through the comments on the proposal, I thought that the configuration system would supply defaults, that could still be overriden by an sdk. 
with the example of `disable_dtx`, should the client implementations assume `dtx: true` if the setting is not present or should they just do nothing in that case and default to whatever is hardcoded as the default?